src/site_upload/powerset_merge/powerset_merge.py

```diff
@@ -118,56 +118,67 @@ def merge_powersets(manager: s3_manager.S3Manager) -> None:
     for latest_path in latest_file_list:
         if manager.version not in latest_path:
             continue
-        site_specific_name = functions.get_s3_site_filename_suffix(latest_path)
+        latest_metadata = functions.parse_s3_key(latest_path)
         subbucket_path = (
-            f"{manager.study}/{manager.study}__{manager.data_package}" f"/{site_specific_name}"
-        )
-        date_str = datetime.datetime.now(datetime.UTC).isoformat()
-        timestamped_name = f".{date_str}.".join(site_specific_name.split("."))
-        timestamped_path = (
-            f"{manager.study}/{manager.study}__{manager.data_package}" f"/{timestamped_name}"
+            f"{manager.study}/{manager.study}__{manager.data_package}/{latest_metadata.site}/" 
+            f"{manager.study}__{manager.data_package}__{manager.version}"
         )
+        archives = []
         try:
             is_new_data_package = False
             # if we're going to replace a file in last_valid, archive the old data
-            if any(x.endswith(site_specific_name) for x in last_valid_file_list):
-                manager.copy_file(
-                    f"{enums.BucketPath.LAST_VALID.value}/{subbucket_path}",
-                    f"{enums.BucketPath.ARCHIVE.value}/{timestamped_path}",
-                )
+            if matches := filter(lambda x: subbucket_path in x, last_valid_file_list):
+                date_str = datetime.datetime.now(datetime.UTC).isoformat()
+                for match in matches:
+                    match_filename = functions.get_s3_filename(match)
+                    match_timestamped_filename = f".{date_str}.".join(match_filename.split("."))
+                    last_target = (
+                        f"{enums.BucketPath.LAST_VALID.value}/{subbucket_path}/{match_filename}"
+                    )
```

Comment:
nit: could this just be `last_target = match`? Just to avoid a mistake when reconstructing the path.

Reply:
yep, sure could